src/components/select/multi-select/multi-select.component.js

```diff
@@ -458,10 +458,10 @@ const MultiSelect = React.forwardRef(
       [textboxRef, actualValue, updateValue]
     );
 
-    function onSelectListClose() {
+    const onSelectListClose = useCallback(() => {
       setOpenState(false);
       setFilterText("");
-    }
+    }, []);
```

Comment:
could you explain why do we need to have useCallback here?

Reply:
`onSelectListClose` method in `MultiSelect` is passed down to `SelectList` where is referenced inside of a `handleEscapeKey` method. The `handleEscapeKey` is then passed down to `useModalManager` which references it inside of `handleClose` method which is then added to `keyup` event listener. Not wrapping it in `useCallback` would result in events being added and removed on each rerender.